lib/extensions/duplicable/active_record/base.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+module Extensions::Duplicable::ActiveRecord::Base
+  module ClassMethods
+    # This function should be declared in model, to make it duplicable.
+    def acts_as_duplicable
+      # Overridden by the model to duplicate its children.
+      #
+      # @param _duplicator [Duplicator] Duplicator object for checking if objects
+      #   have been duplicated.
+      # @param _other [Object] Source object whose children are to be duplicated.
+      define_method :initialize_duplicate do |_duplicator, _other|
+        raise 'Implement your own initialize_duplicate method for this model.'
```

Comment:
@fonglh Think you should use `define_method(:initialize_duplicate)`, it's not a DSL, so use it with brackets.